dao/src/main/java/hu/uni/miskolc/iit/softwaretesting/dao/BookDaoXMLImpl.java

```diff
@@ -37,9 +41,22 @@
      */
     private File outputFile;
 
-    public BookDaoXMLImpl(Document document, File outputFile) {
+    public BookDaoXMLImpl(File inputFile, File outputFile) {
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder = null;
+        try {
+            dBuilder = dbFactory.newDocumentBuilder();
+            this.document = dBuilder.parse(inputFile);
+        } catch (ParserConfigurationException e) {
```

Comment:
Please put these exceptions to the constructor's signature, because it's not logical to instantiate the class if any of these these errors occured.